src/memory.rs

```diff
@@ -55,6 +55,9 @@ impl Pointer {
     pub fn points_to_zst(&self) -> bool {
         self.alloc_id == ZST_ALLOC_ID
     }
+    pub fn to_int(&self) -> usize {
+        self.offset
```

Comment:
A normal use of a pointer would just access `ptr.offset`. I'm thinking of `to_int()` as a way to access the offset while also asserting that the `alloc_id` is `ZST_ALLOC_ID`. So I would write the body as `{ assert!(self.points_to_zst()); self.offset }`.